src/components/ClusterManager.js

```diff
@@ -209,22 +212,22 @@ export default class ClusterManager extends PureComponent {
     const { namespace } = this.props
     this.executeAndRefresh(
       Ajax().Clusters.cluster(namespace, Utils.generateClusterName()).create({
-        machineConfig: normalizeMachineConfig({})
+        runtimeConfig: formatRuntimeConfig(normalizeRuntimeConfig({}))
```

Comment:
While this technically does the right thing, I think there might be a conceptual bug lurking. We essentially have three different types of data describing machines:
* L: the native Leo format, with `cloudService`, and the split sub-fields, e.g. `machineType` vs `masterMachineType`
* N: the normalized format, with `cloudService`, and the sub-fields merged into the dataproc names, e.g. `masterMachineType`
* S: the state format, with `cloudService` and `numWorkers` denormalized as `sparkMode`, and also merged sub-fields

`normalizeRuntimeConfig` is `L => N`, and formatRuntimeConfig is currently `S => L`, so they're technically not compatible. It happens to work, because `formatRuntimeConfig` looks for `sparkMode`, finds it `undefined`, and defaults `cloudService` to `'GCE'`. But I could see this leading to potential confusion or bugs later.

I think a good solution would be to change `formatRuntimeConfig` to `N => L` by having it look at `cloudService` instead, and write an inline `S => N` transform in `getRuntimeConfig`. That keeps all handling of `S` confined to `NewClusterModal`, which makes sense since that's where it's relevant.